### PR TITLE
Added IFrame changes to editor

### PR DESCRIPTION
**Pull request by qwertynerd97**
Built GUI for the new IFrame changes in dev/4-amethyst.  Also merged dev/4-amethyst changes into the editor

Resolves https://github.com/ucfopen/Obojobo/issues/616
Resolves https://github.com/ucfopen/Obojobo/issues/751

### CODE REVIEW

**Comment by gitnix** [comment]
Made a new draft with `test-object.json` from master. After navigating to IFrame Examples, app does not change routes.

File: packages/obojobo-document-engine/ObojoboDraft/Chunks/IFrame/editor-component.js [hunk 1]
```diff
@@ -1,74 +1,275 @@
...
+
+		content[property] = event.target.value
+
+		change.setNodeByKey(this.props.node.key, { data: { content } })
+		editor.onChange(change)
 	}
 
-	renderFrame() {
-		const { isFocused } = this.props
+	handleControlChange(property, checked) {
```
Comment (zachberry) [inline packages/obojobo-document-engine/ObojoboDraft/Chunks/IFrame/editor-component.js]: Preference:

```javascript
	handleControlChange(property, checked) {
		const editor = this.props.editor
		const change = editor.value.change()
		const content = this.props.node.data.get('content')

		const controls = new Set(content.controls.split(','))

		if (checked) {
			controls.add(property)
		} else {
			controls.delete(property)
		}

		content.controls = [...controls].join(',')

		change.setNodeByKey(this.props.node.key, { data: { content } })
		editor.onChange(change)
	}
```

File: packages/obojobo-document-engine/ObojoboDraft/Chunks/IFrame/editor-component.js [hunk 2]
```diff
@@ -1,74 +1,275 @@
...
 		}
 
 		return (
-			<div className={'obojobo-draft--chunks--iframe viewer'} style={wrapperStyle}>
-				<div style={maskStyle} />
+			<div className={'iframe-container'}>
 				<iframe
-					is
+					ref="iframe"
+					title={'banana'}
```
Comment (zachberry) [inline packages/obojobo-document-engine/ObojoboDraft/Chunks/IFrame/editor-component.js]: This is **a _peel_ ing** but probably need a different title

File: packages/obojobo-document-engine/ObojoboDraft/Chunks/IFrame/editor-component.js [hunk 3]
```diff
@@ -1,74 +1,275 @@
 import React from 'react'
+import Common from 'Common'
```
Comment (zachberry) [inline packages/obojobo-document-engine/ObojoboDraft/Chunks/IFrame/editor-component.js]: Similar to https://github.com/ucfopen/Obojobo/pull/671  this file has `id`s that need more specific selectors

**Comment by qwertynerd97** [comment]
Merged https://github.com/ucfopen/Obojobo/pull/671 to resolve an issue with the input boxes that would conflict with https://github.com/ucfopen/Obojobo/pull/714

https://github.com/ucfopen/Obojobo/pull/671 should be merged into dev/5 first

**Comment by qwertynerd97** [comment]
Also resolves https://github.com/ucfopen/Obojobo/issues/751

**Comment by zachberry** [comment]
Seeing some weird graphical issues.

The background expands past the frame of the iframe. I tweaked this image in photoshop as it's hard to see on some monitors:

Screenshot 2019-03-25 10 32 25

Also, previewing a page which just has the text "ok" with no background results in this:

Screenshot 2019-03-25 10 49 34

The preview page seems to be shifted from the edge and the reload button is in a weird place.

Compare to the same page in the viewer (the viewer has a default gray background if the page has no background. We probably should change that to white - I had never tested a background-less page before, but that's why it's gray):

Screenshot 2019-03-25 10 49 27

**Comment by zachberry** [comment]
Initial is misspelled in the UI and in the code, causing it not to work. "initalZoom" should be "initialZoom".

I have some quick fixes for some of the styling issues that I can commit, but before I push I wanna run it by you and make sure I won't break anything.

What do you think about disabling the in-editor preview for now and adding it as a future feature? The IFrame code in the viewer is pretty complicated and I think just dropping in the viewer's IFrame component as a preview would be the way to go (if this is possible).

File: packages/obonode/obojobo-chunks-iframe/iframe-properties-modal.js [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+	}
+
+	handleZoomChange(event) {
+		const initialZoom = event.target.value
+
+		return this.setState({ initialZoom })
+	}
+
+	handleAutoloadChange(checked) {
+		return this.setState({ border: checked })
```
Comment (zachberry) [inline packages/obonode/obojobo-chunks-iframe/iframe-properties-modal.js]:
```suggestion
		return this.setState({ autoload: checked })
```

**Comment by zachberry** [comment]
@gitnix Latest update should address styling concerns

**Pull request merged**